snips_nlu/utils.py

```diff
@@ -223,3 +226,35 @@ def ranges_overlap(lhs_range, rhs_range):
     else:
         raise TypeError("Cannot check overlap on objects of type: %s and %s"
                         % (type(lhs_range), type(rhs_range)))
+
+
+def elapsed_since(time):
+    return datetime.now() - time
+
+
+def json_debug_string(dict_data):
+    return json.dumps(dict_data, ensure_ascii=False, indent=2)
+
+
+def log_result(logger, level, output_msg=None):
+    if output_msg is None:
+        output_msg = "Computed  result in {elapsed_time}:\n{result}"
+
+    def get_wrapper(fn):
+        def wrapped(*args, **kwargs):
+            start = datetime.now()
+            res = fn(*args, **kwargs)
+            try:
+                res_debug_string = json_debug_string(res)
+            except TypeError:
+                res_debug_string = str(res)
+            msg_fmt = {
+                "elapsed_time": datetime.now() - start,
+                "result": res_debug_string
+            }
+            logger.log(level, output_msg.format(**msg_fmt))
```

Comment:
`elapsed_time` and `result` should be computed only when necessary, i.e. when included in `output_msg`.